apps/api/src/profiles/profiles.service.ts

```diff
@@ -1,27 +1,171 @@
-import { Injectable } from '@nestjs/common';
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { PrismaClientUnknownRequestError } from '@prisma/client/runtime';
 import { CreateProfileDto } from './dto/create-profile.dto';
+import { PrismaService } from '../prisma/prisma.service';
 import { UpdateProfileDto } from './dto/update-profile.dto';
-import { Profile } from './entities/profile.entity';
+
+interface Resp {
+  message: string;
+  data?: unknown;
+}
 
 @Injectable()
 export class ProfilesService {
-  create(createProfileDto: CreateProfileDto): Profile {
-    return { name: createProfileDto.name };
+  constructor(private prismaService: PrismaService) {}
+
+  // Response Handler
+  Success(resp: Resp) {
+    return {
+      Success: true,
+      message: resp.message,
+      data: resp.data,
+    };
   }
 
-  findAll(): Profile[] {
-    return [{ name: 'test' }];
+  // Method to CREATE a new profile
+  async create(createProfileDto: CreateProfileDto) {
+    const ReadResp = await this.read(createProfileDto.authid);
+    const EmailResp = await this.EmailRead(createProfileDto.email);
+    if (ReadResp.data != null) {
+      throw new HttpException(
+        {
+          success: false,
+          error: 'User Exists',
+        },
+        HttpStatus.CONFLICT
+      );
+    }
+    if (EmailResp != null && EmailResp.data != null) {
+      throw new HttpException(
+        {
+          success: false,
+          error: 'Email already in use',
+        },
+        HttpStatus.CONFLICT
+      );
+    }
+    const d = new Date(createProfileDto.dob);
+    const createProfile: CreateProfileDto = createProfileDto;
+    createProfile.dob = d;
+    try {
+      const resp = await this.prismaService.user.create({
+        // @ts-ignore
+        data: createProfileDto,
+      });
+      return this.Success({
+        data: resp,
+        message: 'User was created succesfully',
+      });
+    } catch (err) {
+      if (err instanceof PrismaClientUnknownRequestError) {
+        throw err;
+      } else {
+        throw new HttpException(
+          {
+            success: false,
+            error: "Could'nt create user",
+          },
+          HttpStatus.BAD_REQUEST
+        );
+      }
+    }
   }
 
-  findOne(id: number): Profile {
-    return { name: `test ${id}` };
+  // Method to READ an existing profile
+  async read(authid: string) {
+    try {
```

Comment:
You can remove these try catch and let it fail as its handled in controller layer